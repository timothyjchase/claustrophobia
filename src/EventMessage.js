import React from 'react'
import { observer } from 'mobx-react'
import { Dropdown, Message } from 'semantic-ui-react'
import renderHTML from 'react-render-html'

const EventMessage = ({ game }) => {
  if (game.event) {
    return (
      <Message negative>
        <div style={{ float: 'right' }}>
          <Dropdown text="" error>
            <Dropdown.Menu direction="left">
              <Dropdown.Item
                text="Cancel Event"
                onClick={() => game.removeEvent(true)}
              />
            </Dropdown.Menu>
          </Dropdown>
        </div>
        <Message.Header>{game.event.name}</Message.Header>
        <Message.Content>
          {renderHTML(game.event.description || '')}
        </Message.Content>
      </Message>
    )
  }
  return (
    <Message negative>
      <Message.Content>Event cancelled</Message.Content>
    </Message>
  )
}

export default observer(EventMessage)
