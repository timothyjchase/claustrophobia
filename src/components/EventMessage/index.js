import React from 'react'
import PropTypes from 'prop-types'
import { Dropdown, Message } from 'semantic-ui-react'
import renderHTML from 'react-render-html'

const EventMessage = ({ upcomingEvent, removeEvent }) => (
  <Message negative>
    <div style={{ float: 'right' }}>
      <Dropdown text="" error>
        <Dropdown.Menu direction="left">
          <Dropdown.Item text="Cancel Event" onClick={removeEvent} />
        </Dropdown.Menu>
      </Dropdown>
    </div>
    <Message.Header>{upcomingEvent.name}</Message.Header>
    <Message.Content>
      {renderHTML(upcomingEvent.description || '')}
    </Message.Content>
  </Message>
)

EventMessage.propTypes = {
  upcomingEvent: PropTypes.object.isRequired,
  removeEvent: PropTypes.func.isRequired,
}

export default EventMessage
