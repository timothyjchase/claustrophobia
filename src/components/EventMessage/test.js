import React from 'react'
import EventMessage from './index'

describe('<EventMessage />', () => {
  const commonProps = {
    upcomingEvent: { name: 'Test event', description: 'Test description' },
    removeEvent: jest.fn(),
  }
  describe('with test event', () => {
    const component = shallow(<EventMessage {...commonProps} />)
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
