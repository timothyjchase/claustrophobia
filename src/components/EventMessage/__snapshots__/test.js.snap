// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventMessage /> with test event should match snapshot 1`] = `
<Message
  negative={true}
>
  <div
    style={
      Object {
        "float": "right",
      }
    }
  >
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      deburr={false}
      error={true}
      icon="dropdown"
      minCharacters={1}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      searchInput="text"
      selectOnBlur={true}
      selectOnNavigation={true}
      text=""
      wrapSelection={true}
    >
      <DropdownMenu
        direction="left"
      >
        <DropdownItem
          onClick={[MockFunction]}
          text="Cancel Event"
        />
      </DropdownMenu>
    </Dropdown>
  </div>
  <MessageHeader>
    Test event
  </MessageHeader>
  <MessageContent>
    Test description
  </MessageContent>
</Message>
`;
